### PR TITLE
[dom-gpu] NAMD and CHARM easyconfig for PE 17.08

### PR DESCRIPTION
**Pull request by ChristopherBignamini**
Could you please have a look? Thanks!

### CODE REVIEW

**Comment by gppezzi** [comment]
retest this please

**Comment by lucamar** [comment]
retest this please

**Comment by lucamar** [comment]
I will use this pull request to test the new Jenkins project `TestingEB`, so please don't be worried if it fails

**Comment by lucamar** [comment]
retest this please

**Comment by lucamar** [comment]
`ERROR: Irresolvable dependencies encountered: Charm++/6.7.1-CrayIntel-17.08`

**Comment by ChristopherBignamini** [comment]
Error should be resolved by the last commit

**Comment by lucamar** [comment]
```
Current diff list by 'git diff origin/master..HEAD --name-only --oneline --no-merges --diff-filter=ACMRTUXB':
easybuild/cray_external_modules_metadata.cfg
easybuild/easyconfigs/c/Charm++/Charm++-6.7.1-CrayIntel-17.08.eb
easybuild/easyconfigs/c/Charm++/Charm++-6.8.0-CrayIntel-17.08.eb
easybuild/easyconfigs/n/NAMD/NAMD-2.12-CrayIntel-17.08-cuda-8.0.eb
easybuild/easyconfigs/n/NAMD/NAMD-2.12-CrayIntel-17.08.eb
easybuild/easyconfigs/n/NAMD/NAMD-2.12.patch
easybuild/easyconfigs/v/VMD/VMD-1.9.3-egl.eb
easybuild/easyconfigs/v/VMD/VMD-1.9.3-ogl.eb
easybuild/easyconfigs/v/Visit/Visit-2.12.0-CrayGNU-2016.11.eb
```

**Comment by lucamar** [comment]
Can you please try if NAMD can be built using `Charm++-6.8.0-changa-CrayIntel-17.08-cuda-8.0.eb` and `Charm++-6.8.0-changa-CrayIntel-17.08.eb` that are already available in production in `17.08`?

**Comment by ChristopherBignamini** [comment]
@lucamar I'm using Charm 6.7.1 because, according to the release notes of NAMD 2.12, it seems that this is the foreseen version. I added Charm 6.8.0 because @jgphpc asked it for another module.

**Comment by jgphpc** [comment]
I will remove 6.8 as it is already in master

**Comment by jgphpc** [comment]
:ok: https://jenkins.cscs.ch/job/TestingEB/label=dom/600/console

File: easybuild/easyconfigs/c/Charm++/Charm++-6.7.1-CrayIntel-17.08.eb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Contributed by gppezzi, Luca Marsella and Christopher Bignamini (CSCS)
+easyblock = 'CmdCp'
+
+nameletterlower = 'c'
```
Comment (lucamar) [inline easybuild/easyconfigs/c/Charm++/Charm++-6.7.1-CrayIntel-17.08.eb]: You don't need to define this variable, as it is already created by EasyBuild: 
```
$ eb --avail-easyconfig-templates | grep nameletterlower
    %(nameletterlower)s: lower case of value of nameletter
    PYPI_LOWER_SOURCE: pypi source url (lowercase name) (https://pypi.python.org/packages/source/%(nameletterlower)s/%(namelower)s)
```

File: easybuild/easyconfigs/n/NAMD/NAMD-2.12-CrayIntel-17.08-cuda-8.0.eb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Contributed by gppezzi, Luca Marsella and Christopher Bignamini (CSCS)
+easyblock = 'MakeCp'
+
+nameletterlower = 'n'
```
Comment (lucamar) [inline easybuild/easyconfigs/n/NAMD/NAMD-2.12-CrayIntel-17.08-cuda-8.0.eb]: You don't need to define this variable, as it is already created by EasyBuild:
```
$ eb --avail-easyconfig-templates | grep nameletterlower
    %(nameletterlower)s: lower case of value of nameletter
```

**Comment by lucamar** [comment]
I think we should try to build using up to date versions of the dependencies, as far as possible. 

`NAMD` developers package NAMD give their own Charm release, in this case `6.7.1`, but we might give `6.8.0` a try anyway. 

Same holds for `Tcl`, which is not `8.6.7` while `NAMD` packages `8.5.9`.

I know that it takes some time to test these build, but that's the goal of the milestone `Summer update`.

**Comment by jgphpc** [comment]
> give 6.8.0 a try

In that case, please try using the charm++ from master: 
https://github.com/eth-cscs/production/tree/master/easybuild/easyconfigs/c/Charm%2B%2B

**Comment by lucamar** [comment]
I have modified the recipes using the latest modules available in production: I have used `Charm++` without cuda support, as the NAMD packaged CHamr version was not using it, however we can re-assess that if necessary.

I only miss a patch to `arch/CRAY-XC.tcl` in order to remove `-ltcl8.5` and use the current Tcl version instead.

**Comment by lucamar** [comment]
I have updated the patch `NAMD-2.12.patch` in order to match Tcl release `8.6` when linking and the build works for both cuda and non cuda versions in my environment.

**Pull request merged**